django/db/backends/base/operations.py

```diff
@@ -260,6 +260,17 @@ def limit_offset_sql(self, low_mark, high_mark):
             if sql
         )
 
+    def fk_on_delete_sql(self, operation):
+        """
+        Return the SQL to make an ON DELETE statement during a CREATE TABLE
+        statement.
+        """
+        if operation in ["CASCADE", "SET NULL", "RESTRICT"]:
```

Comment:
What about `"NO ACTION"` and `"SET DEFAULT"`? See:

- https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-FK
- https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html
- https://docs.oracle.com/en/database/oracle/oracle-database/23/sqlrf/constraint.html#GUID-1055EA97-BA6F-4764-A15F-1024FD5B6DFE__GUID-8E388631-3D99-4445-A2CD-A8776E8CEFC1
- https://www.sqlite.org/foreignkeys.html#fk_actions

Looks like Oracle is the only one that wouldn't support these. (And notably, it also doesn't look like it supports `"RESTRICT"`.

For `"SET DEFAULT"` we'd also need to ensure that `db_default` is set on the field. (Recently added to Django 5.0)

Reply:
Thank you for the review Nick. Will try to add the `DB_SET_DEFAULT` in the next commit and add a warning message for oracle database, in that case we may redirect to django based `RESTRICT`.

Reply:
In the original PR, I think we left out NO_ACTION since it's the default FK behavior--even if you don't add an `ON DELETE...` statement. It's likely equivalent to django's DO_NOTHING. It wouldn't hurt to add it for completeness, but it might also add unnecessary options that do the same thing. Seems like a style preference. 

@ngnpope how sure are you about Oracle not supporting RESTRICT? Im not an oracle expert, but the language linked makes it seem like only _views_ dont support RESTRICT (but maybe tables would?).

> Restriction on ON DELETE
> You cannot specify this clause for a view constraint.

Reply:
> In the original PR, I think we left out NO_ACTION since it's the default FK behavior--even if you don't add an ON DELETE... statement. It's likely equivalent to django's DO_NOTHING. It wouldn't hurt to add it for completeness, but it might also add unnecessary options that do the same thing. Seems like a style preference.

My concern was that `DO_NOTHING` makes no changes to the database. So if you've set, e.g. `DB_RESTRICT` and change to `DO_NOTHING` then the database would still be using `RESTRICT` because no migration would be generated? Because there is nothing to `ALTER` it to `NO ACTION`? I haven't had time to investigate this in detail - it's a hunch - but we should have tests for all transitions between database-managed and "traditional" `on_delete` options.

This is also why I've thought it might be cleaner to have a `db_on_delete` like we have a `db_default`... We need to get this right - not rush - as we'll be stuck with the decision.

Reply:
Good point on migrations. It's been a while since Ive used django.

IIRC, the migrations aren't persisted as raw SQL though, right? So wouldnt changing the kwarg to DO_NOTHING possibly still *serialize* differently into the migration file than DB_CASCADE? And if it serializes differently, wont that then be able to generate new SQL for the table? I guess it's possible that wouldnt be backwards compatible though--if someone already has migrations manually adding CASCADE and they put DO_NOTHING in their python code...

Okay, thinking about it, I guess DB_NO_ACTION makes sense 👍 .

Reply:
> Best thing is to ensure that there are comprehensive tests migrating between all possible options.

Yes, we need migration tests.

Reply:
Added tests for migrations.